### PR TITLE
Added existing Code

### PR DESCRIPTION
**Pull request by torbrenner**

### CODE REVIEW

File: auth/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# LensAuth
```
Comment (patrick-skowronek) [inline auth/README.md:1]: Readme could be improved

File: auth/src/lib/user.service.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
+import {Inject, Injectable} from '@angular/core';
+import {AuthConfig, OAuthService} from "angular-oauth2-oidc";
+import {BehaviorSubject} from "rxjs";
+import {LENS_AUTH_CONFIG_TOKEN, LensAuthConfig} from "./lens-auth-config";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
+  // TODO: Is this still needed?
```
Comment (patrick-skowronek) [inline auth/src/lib/user.service.ts]: Resolve todo

File: auth/src/lib/user.service.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    }
+  }
+
+  /**
+   * Handles the login process of the user.
+   * @returns Promise<boolean>, that specifies whether the user is logged in.
+   */
+  login(): Promise<boolean> {
+    return this.oauthService.loadDiscoveryDocumentAndTryLogin().then(() => {
+      // TODO: Check if we can do
```
Comment (patrick-skowronek) [inline auth/src/lib/user.service.ts]: Resolve todo

File: auth/src/lib/user.service.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        this.oauthService.initLoginFlow();
+        return false;
+      } else {
+        let authorized = this.isAuthorized();
+        this.isAuthenticatedSubject$.next(authorized)
+        return authorized;
+      }
+    }).catch(() => {
+      console.error("Failed to load discovery document from identity provider.")
+      return false
```
Comment (patrick-skowronek) [inline auth/src/lib/user.service.ts:46]: We should start to think about ending a line with ";"

File: components/README.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# LensComponents
```
Comment (patrick-skowronek) [inline components/README.md:1]: Add description to readme

File: components/criteria-catalogue/src/boolean-condition-builder/boolean-condition-builder.component.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+} from '@angular/core';
+import {
+  CatalogueService,
+  QueryService,
+  Condition,
+  ConditionTypes,
+  Criteria
+} from '@samply/lens-core';
+
+/*
```
Comment (patrick-skowronek) [inline components/criteria-catalogue/src/boolean-condition-builder/boolean-condition-builder.component.ts]: Can we remove this code? Seems project specific

File: components/criteria-catalogue/src/criteria-catalogue.component.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      this.queryService.delete(condition);
+    } else if (this.queryService.read(condition.key)) {
+      this.queryService.update(condition);
+    } else {
+      this.queryService.create(condition);
+    }
+  }
+
+  onOperationChange(operation: Operation) {
+    // TODO: Implement check for existing operation
```
Comment (patrick-skowronek) [inline components/criteria-catalogue/src/criteria-catalogue.component.ts]: Resolve todo

File: components/criteria-catalogue/src/criteria-catalogue.component.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    console.log("OperationChanged triggered!")
+    if (true) {
+      this.queryService.create(operation)
+    } /*else if (false) {
```
Comment (patrick-skowronek) [inline components/criteria-catalogue/src/criteria-catalogue.component.ts]: Can we remove this?

File: components/criteria-catalogue/src/criteria-catalogue.component.ts [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    } else if (this.queryService.read(condition.key)) {
+      this.queryService.update(condition);
+    } else {
+      this.queryService.create(condition);
+    }
+  }
+
+  onOperationChange(operation: Operation) {
+    // TODO: Implement check for existing operation
+    console.log("OperationChanged triggered!")
```
Comment (patrick-skowronek) [inline components/criteria-catalogue/src/criteria-catalogue.component.ts]: Please remove console.log and replace it with log.debug

File: components/criteria-catalogue/src/criteria-catalogue.module.ts [hunk 10]
```diff
@@ -0,0 +1,52 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { CriteriaCatalogueComponent } from './criteria-catalogue.component';
+import {TreeModule} from "primeng/tree";
```
Comment (patrick-skowronek) [inline components/criteria-catalogue/src/criteria-catalogue.module.ts:4]: Imports are not consistent

File: components/criteria-catalogue/src/date-condition-builder/date-condition-builder.component.ts [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+} from '@angular/core';
+import {
+  CatalogueService,
+  QueryService,
+  Condition,
+  Criteria
+} from '@samply/lens-core';
+
+/*
+* This component lets the user define conditions based on dates.
```
Comment (patrick-skowronek) [inline components/criteria-catalogue/src/date-condition-builder/date-condition-builder.component.ts]: Can we move this into the documentation?

File: components/criteria-catalogue/src/date-condition-builder/date-condition-builder.component.ts [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+          this.criteria.allowedConditionTypes[0],
+          this.criteria.system,
+          {min: undefined, max: undefined},
+        )
+      }
+    })
+  }
+
+  onChange() {
+    // TODO: onInput would fire without pressing "ENTER", search a way for updating ngModel according to new value passed as $event.value
```
Comment (patrick-skowronek) [inline components/criteria-catalogue/src/date-condition-builder/date-condition-builder.component.ts]: Resolve todo

File: components/criteria-catalogue/src/number-condition-builder/number-condition-builder.component.css [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+  flex-flow: row nowrap;
+  align-items: baseline;
+  gap: 0.5rem;
+}
+::ng-deep .p-inputnumber-from input, ::ng-deep .p-inputnumber-to input {
+  font-size: 12pt;
+  padding: 0.3rem 0.5rem;
+}
+
+/*
```
Comment (patrick-skowronek) [inline components/criteria-catalogue/src/number-condition-builder/number-condition-builder.component.css]: Do we need this code?

File: components/criteria-catalogue/src/number-condition-builder/number-condition-builder.component.ts [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+  Output
+} from '@angular/core';
+import {
+  CatalogueService,
+  QueryService,
+  Condition,
+  Criteria
+} from '@samply/lens-core';
+
+/*
```
Comment (patrick-skowronek) [inline components/criteria-catalogue/src/number-condition-builder/number-condition-builder.component.ts]: Can we move this into the documentation?

File: components/criteria-catalogue/src/number-condition-builder/number-condition-builder.component.ts [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+          this.criteria.allowedConditionTypes[0],
+          this.criteria.system,
+          (this.criteria.allowedConditionTypes[0] == "EQUALS") ? 0 : {min: 0, max: 0}
+        )
+      }
+    })
+  }
+
+  onChange() {
+    // TODO: onInput would fire without pressing "ENTER", search a way for updating ngModel according to new value passed as $event.value
```
Comment (patrick-skowronek) [inline components/criteria-catalogue/src/number-condition-builder/number-condition-builder.component.ts]: Resolve todo

File: components/criteria-catalogue/src/string-condition-builder/string-condition-builder.component.ts [hunk 16]
```diff
@@ -0,0 +1,233 @@
...
+
+  computeWildcardSuggestions(currentSuggestions: Array<{key: string, de: string, en: string}>, currentSearch: string): Array<{key: string, de: string, en: string}> {
+    let wildcardSuggestions: Array<{key: string, de: string, en: string}> = [];
+    currentSuggestions.forEach(child => {
+      if (!wildcardSuggestions.some(
+        code => code.key.startsWith(child.key.split(".")[0])
+      )) {
+        wildcardSuggestions.push({
+          key: child.key,
+          // TODO: We could accumulate all merged codes here to add a hint on all codes included.
```
Comment (patrick-skowronek) [inline components/criteria-catalogue/src/string-condition-builder/string-condition-builder.component.ts]: Resolve todo

File: components/loading-bar/src/loading-bar.component.css [hunk 17]
```diff
@@ -0,0 +1,8 @@
+::ng-deep .p-progressbar {
+  height: 15px;
+}
+
+::ng-deep .p-progressbar-value {
+  /*TODO: Find a way without important */
```
Comment (patrick-skowronek) [inline components/loading-bar/src/loading-bar.component.css]: Resolve todo

File: components/loading-bar/src/loading-bar.component.ts [hunk 18]
```diff
@@ -0,0 +1,57 @@
+import {
+  Component,
+  Inject
+} from '@angular/core';
+import {
+  LensConfig,
+  LENS_CONFIG_TOKEN,
+  QueryService
+} from '@samply/lens-core';
+// TODO: The import of this submodule doesn't seem to work, needs further invesitagtion
```
Comment (patrick-skowronek) [inline components/loading-bar/src/loading-bar.component.ts:10]: Resolve todo

File: components/result-summary-bar/src/result-summary-bar.component.html [hunk 19]
```diff
@@ -0,0 +1,9 @@
+<h1>{{resultRenderer.title}}</h1>
+<div class="flex align-items-center flex-wrap">
+  <!-- TODO: Add a configuration for the pictogram mapping -->
```
Comment (patrick-skowronek) [inline components/result-summary-bar/src/result-summary-bar.component.html]: Resolve todo

File: components/search-bar/src/search-bar.component.ts [hunk 20]
```diff
@@ -0,0 +1,147 @@
...
+
+  isOperation(item: any): boolean {
+    return item instanceof Operation;
+  }
+
+  suggestionSelected($event: Condition | Operation) {
+    let existingValue = this.queryService.read($event.key);
+    console.log(existingValue)
+    if (existingValue) {
+      // TODO: This needs to merge with the previous values!!!
```
Comment (patrick-skowronek) [inline components/search-bar/src/search-bar.component.ts]: Resolve todo

File: core/cql/src/beam.ts
Comment (patrick-skowronek) [inline core/cql/src/beam.ts:1]: Will create Issue for this file

File: auth/src/lib/user.service.ts (hunk 4 above)
Comment (torbrenner) [reply auth/src/lib/user.service.ts:46]: I will create an issue for activating linting. With activated linting, this will be checked.

Comment (torbrenner) [reply auth/src/lib/user.service.ts:46]: #4

File: components/criteria-catalogue/src/criteria-catalogue.module.ts (hunk 10 above)
Comment (torbrenner) [reply components/criteria-catalogue/src/criteria-catalogue.module.ts:4]: should resolve with #4

**Review by patrick-skowronek — APPROVED** [review]

**Pull request merged**